connectors/sources/sharepoint_online.py

```diff
@@ -1299,29 +1407,36 @@ async def get_docs(self, filtering=None):
                 site_collection["siteCollection"]["hostname"],
                 self.configuration["site_collections"],
             ):
-                access_control = await self._site_access_control(site)
-                yield self._decorate_with_access_control(site, access_control), None
+                site_access_control = await self._site_access_control(site)
+                yield self._decorate_with_access_control(
+                    site, site_access_control
+                ), None
 
                 async for site_drive in self.site_drives(site):
                     yield self._decorate_with_access_control(
-                        site_drive, access_control
+                        site_drive, site_access_control
```

Comment:
Can drives not be given more granular access than their parent site?

Reply:
They're treated as lists AFAIU and they can have unique permissions. Will be addressed in a separate PR 👍 Issue for tracking: https://github.com/elastic/enterprise-search-team/issues/5363